lessons/ro/chapter_8.yaml

```
@@ -0,0 +1,357 @@
+- title: Capitolul 8 - Pointeri "inteligenți"
+  content_markdown: >
+    În acest capitol vom lămuri pointerii inteligenți. Să explorăm aceste structuri
+    
+    de date care ne permit să interacționăm cu memoria la cel mai de bază nivel.
+
+
+    Ferris spune: "Nu te simții copleșit de acest capitol dacă simți că
+    
+    nu poți gestiona de unul singur memoria la un nivel de bază după această
+    
+    lectură scurtă. Scopul acestui capitol este să îți prezinte câteva unelte
+    
+    și să îți creezi o idee despre cum se folosesc!"
+- title: Să revizuim referințele
+  content_markdown: >
+    O referință este doar un număr ce marchează începutul unei zone în memorie.
+    
+    Singurul său scop este să reprezinte conceptual locul în care se află un 
+    
+    anumit tip de dată. Ceea ce deosebește o referință de un simplu număr este
+    
+    faptul că limbajul Rust se asigură că durata de viață a unei referințe nu este
+    
+    mai mare decât cea a valorii referențiate (altfel ar apărea erori când o folosim).
+- title: Pointeri simpli
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=fn%20main()%20%7B%0A%20%20%20%20let%20a%20%3D%2042%3B%0A%20%20%20%20let%20memory_location%20%3D%20%26a%20as%20*const%20i32%20as%20usize%3B%0A%20%20%20%20println!(%22Data%20is%20here%20%7B%7D%22%2C%20memory_location)%3B%0A%7D%0A
+  content_markdown: >
+    Referințele pot fi transformate într-un tip de dată mai simplu, 
+    
+    numit *pointer simplu*. La fel ca un număr, acesta poate fi copiat si mutat
+    
+    fără prea multe restricții. Limbajul Rust nu ne asigură cu nimic de 
+    
+    valabilitatea zonei de memorie la care pointează.
+
+
+    Există două feluri de pointeri simpli:
+
+
+    * `*const T` - Un pointer simplu către un tip de dată T care nu va fi modificată.
+
+    * `*mut T` - Un pointer simplu către un tip de dată T ce poate fi modificată.
```

```suggestion
    * `*mut T` - Un pointer simplu către un tip de date T ce poate fi modificat.
```
Tip de date în locul "tip de dată", iar verbul "a modifica" face referire la un substantiv care nu este la feminin.